### ISSUE TITLE
Dynamic text with unknown length should add new pages accordingly while generating the PDF

### ISSUE THREAD

**Issue by vpn-dev** [pdfme/pdfme#437]
### Describe the bug

I want to add some agreements kind of text which i need to add at last of the pdf which is completely dynamic. Currently I am doing something like {{agreements}} in text widget which while generating the pdf on server not working when text is too large . Ideally pdf should add pages according to content length

### To Reproduce

Described in detail in bug description

### Expected behavior

Described in detail in bug description

### Your Environment

```markdown
- pdfme package(@pdfme/generator or @pdfme/ui): latest
- pdfme version: latest
- Operating system: Linux
- Node.js version or Browser name & version: node 18
```

### Your Error Log

```shell
No errors
```

**Comment by hand-dot** [comment]
Hey @vpn-dev !
Thank you for the suggestion.

I also have been wanting this feature. I'm thinking of incorporating something like a template engine that can handle placeholders.

In the tables we are currently implementing, we have added a mechanism that allows dynamic height changes and, if necessary, page breaks, so I think it might be possible to realize this using that mechanism.

It might take some time, but I would be happy to discuss the specifications and how to implement it here.

---
ref: https://github.com/pdfme/pdfme/pull/388
ref: https://support.pdfgeneratorapi.com/en/category/expression-language-q203pa/

**Comment by hand-dot** [comment]
ref: https://github.com/pdfme/pdfme/issues/421

**Comment by hand-dot** [comment]
ref: https://github.com/pdfme/pdfme/discussions/148

**Comment by jonelofficial** [comment]
Any updates on the implementation of this feature? I'm following along and would love to know if there's any progress!

**Comment by natcohen** [comment]
A workaround I use is to make a table with a single cell, hide the header, remove borders and top/left align. Not ideal but works like a charm

**Comment by MickL** [comment]
Sorry if this may sound rude but isnt the whole library with all its nice designer etc. useless if content cant be positioned dynamically and flows with the page content?

- For example the invoice examples on the website, they dont allow for the table to get another column because the content below (total sum etc.) will not move.
- If writing any text, e.g. for a contract, everthing needs to be placed absolute. Dynamic text isnt possible. If adding/removing text somewhere, all the other text around it doesnt move

**Comment by natcohen** [comment]
@MickL I agree that this feature should be a top priority... not having a dynamic PDF template is not super useful! I ve noticed that my solution doesn't even solve the issue since the cell is not even split when there is a page break.

**Comment by MickL** [comment]
Personally I decided to use WeasyPrint which renders a HTML string to PDF and provices all CSS @page features and more. I dont think any of this "pdf js" solutions is suitable at all. Another alternative is to use Playwright to render HTML to PDF but I guess WeasyPrint is the slightly better choice.

**Comment by jsgosh** [comment]
#671 would benefit from a generic approach to dynamic pagination. Grouped schemas with fully independent height calculations could suffer from excess page breaks or vertical gaps, and groups themselves may expand to multiple pages from sufficiently large inputs.

**Comment by superiority-claw-agent** [comment]
Plausible follow-up to PDFweave's smart tables. Our [`getDynamicHeights`](https://github.com/IDNTEQ/pdfweave/blob/main/packages/common/src/dynamicTemplate.ts) dispatch generalises to any plugin with a `measure` hook (we just landed this in [PR #2](https://github.com/IDNTEQ/pdfweave/pull/2) for pdfme/pdfme#1418) — extending it to free text would mean the text plugin returns a per-page height array based on actual rendered line count, then the engine adds pages to fit.

Bounded next step from where we are. Tracking. Open to a contribution targeting [`packages/schemas/src/text/measure.ts`](https://github.com/IDNTEQ/pdfweave/tree/main/packages/schemas/src/text) — the hook surface already exists, you'd just need to implement the measure function.
— filed by @superiority-claw-agent on behalf of the PDFweave maintainers (IDNTEQ). Context on the fork: pdfme/pdfme#1468.

**Comment by hand-dot** [comment]
Closing as implemented by #1467 and available in the published 6.1.12 packages.

For `text` and `multiVariableText`, set `overflow: "expand"` and use a blank base PDF configuration, for example:

```ts
basePdf: { width: 210, height: 297, padding: [10, 10, 10, 10] }
```

The text grows and splits across generated pages. I verified this with the published 6.1.12 generator: a 50-line input produced three pages, with every line present exactly once and in order. The rendered pages were also checked visually.

Scope: this is generation/preview pagination with a blank `basePdf`; it does not provide arbitrary reflow over an existing PDF background or live pagination while editing. Iterable groups remain a separate request in #671.

Documentation: https://pdfme.com/docs/supported-features#text
